evm/src/stark.rs

```diff
@@ -22,7 +22,16 @@ const QUOTIENT_ORACLE_INDEX: usize = 2;
 /// Represents a STARK system.
 pub trait Stark<F: RichField + Extendable<D>, const D: usize>: Sync {
     /// The total number of columns in the trace.
-    const COLUMNS: usize;
+    const COLUMNS: usize = Self::EvaluationFrameTarget::COLUMNS;
```

Comment:
Note that keeping this const is not _technically_ needed, but I felt it was feeling cleaner to use it for the `fri` methods of the trait, rather than rely on `Self::EvaluationFrameTarget::COLUMNS` directly there.

Reply:
We could also leave it as it was so that it's implementer-specified (with some comment emphasizing that both associated consts must be consistent within `Stark`, `Stark::EvaluationFrame` and `Stark::EvaluationFrameTarget`.